### DISCUSSION TITLE
Do StrongSwan version 5.9.14 have any kernel dependency?

### DISCUSSION THREAD

**Discussion by ARCH0911** [strongswan/strongswan#2218] [Q&A]
Linux  4.1.35-rt41 #1 SMP Tue Dec 5 00:03:43 IST 2023 ppc64 GNU/Linux
Do StrongSwan version  5.9.14 is compatible with above the kernel?

**Comment by tobiasbrunner** [comment]
Difficult to say (and also depends on the kernel options). What problem do you have?

**Reply by ARCH0911** [reply]
I am getting below error
swanctl -i
connecting to 'unix:///var/run/charon.vici' failed: No such file or directory
Error: connecting to 'default' URI failed: No such file or directory
strongSwan 5.9.14 swanctl

And I see ipsec/strongswan commands are also not present.

**Reply by suruifu** [reply]
Is charon running in the background?
You can try `ps aux | grep charon` and paste the output here.

**Reply by gowdamanil** [reply]
Hello @ARCH0911 

I had same issue, this was because charon has not been started, start the charon daemon using 
systemctl start strongswan

If not possible, download charon-systemd via command 
apt-get install charon-systemd


then start  the strongswan with 
systemctl start strongswan

This might be helpful thanks.

**Reply by ARCH0911** [reply]
Hi @gowdamanil
 Thank you for the suggestion, I see the Charon is not running. Will try to start the Charon and update.

**Reply by ARCH0911** [reply]
Hi @gowdamanil 
In other server I configure the strongswan with --enable-systemd flag
and when I start strongswan service I got the below errors:

[root@Linux strongswan-5.9.14]# sudo systemctl status strongswan
● strongswan.service - strongSwan IPsec IKEv1/IKEv2 daemon using swanctl
   Loaded: loaded (/usr/lib/systemd/system/strongswan.service; enabled; vendor preset: disabled)
   Active: failed (Result: start-limit) since Tue 2024-04-23 15:07:23 IST; 4min 49s ago
  Process: 7231 ExecStart=/usr/sbin/charon-systemd (code=killed, signal=SEGV)
 Main PID: 7231 (code=killed, signal=SEGV)
   Status: "Starting charon-systemd, strongSwan 5.9.14, Linux 3.10.0-1160.92.1.el7.x86_64, x86_64"

Apr 23 15:07:23 Linux systemd[1]: Failed to start strongSwan IPsec IKEv1/IKEv2 daemon using swanctl.
Apr 23 15:07:23 Linux systemd[1]: Unit strongswan.service entered failed state.
Apr 23 15:07:23 Linux systemd[1]: strongswan.service failed.
Apr 23 15:07:23 Linux systemd[1]: strongswan.service holdoff time over, scheduling restart.
Apr 23 15:07:23 Linux systemd[1]: Stopped strongSwan IPsec IKEv1/IKEv2 daemon using swanctl.
Apr 23 15:07:23 Linux systemd[1]: start request repeated too quickly for strongswan.service
Apr 23 15:07:23 Linux systemd[1]: Failed to start strongSwan IPsec IKEv1/IKEv2 daemon using swanctl.
Apr 23 15:07:23 Linux systemd[1]: Unit strongswan.service entered failed state.
Apr 23 15:07:23 Linux systemd[1]: strongswan.service failed.

could this error be because of the kernel version?
Please let me know If I needs to follow any other way to install or start the charon
Thank you

**Reply by tobiasbrunner** [reply]
Read the log for why this failed.

**Reply by ARCH0911** [reply]
Hi @tobiasbrunner ,
It is segmentation fault (code=killed, signal=SEGV). I was asking for the reason, could it be because of kernel version or the installation I did got corrupted.
want know how to debug it further.
Thank you!

**Reply by tobiasbrunner** [reply]
That has most likely nothing to do with the kernel. It's rather an issue with mixing plugins, libraries and executables from different versions. If you build from sources, always make sure to remove all distribution packages and binaries from previous versions.
